common/persistence/pinotVisibilityTripleManager.go

```diff
@@ -290,91 +300,279 @@ func (v *pinotVisibilityTripleManager) chooseVisibilityManagerForWrite(ctx conte
 	}
 }
 
+// For Pinot Migration uses. It will be a temporary usage
+type userParameters struct {
+	operation    string
+	isOpen       bool
+	domainName   string
+	workflowType string
+	workflowID   string
+	closeStatus  int
+	customQuery  string
+}
+
+// For Pinot Migration uses. It will be a temporary usage
+// logUserQueryParameters will log user queries' parameters so that a comparator workflow can consume
+func (v *pinotVisibilityTripleManager) logUserQueryParameters(userParam userParameters, domain string) {
+	// Don't log if it is not enabled
+	if !v.logCustomerQueryParameter(domain) {
+		return
+	}
+	randNum := rand.Intn(10)
+	if randNum != 5 { // Intentionally to have 1/10 chance to log custom query parameters
+		return
+	}
+
+	v.logger.Info("Logging user query parameters for Pinot/ES response comparator...",
+		tag.OperationName(userParam.operation),
+		tag.IsWorkflowOpen(userParam.isOpen),
+		tag.WorkflowDomainName(userParam.domainName),
+		tag.WorkflowType(userParam.workflowType),
+		tag.WorkflowID(userParam.workflowID),
+		tag.WorkflowCloseStatus(userParam.closeStatus),
+		tag.VisibilityQuery(userParam.customQuery))
+}
+
 func (v *pinotVisibilityTripleManager) ListOpenWorkflowExecutions(
 	ctx context.Context,
 	request *ListWorkflowExecutionsRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:    "list",
+		isOpen:       true,
+		domainName:   request.Domain,
+		workflowType: "",
+		workflowID:   "",
+		closeStatus:  math.MinInt, // set the default close status to be Integer.MIN_VALUE
+		customQuery:  "",
+	}, request.Domain)
+
 	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	if override := ctx.Value("visibility-override"); override == Primary {
+		manager = v.esVisibilityManager
+	} else if override == Secondary {
+		manager = v.pinotVisibilityManager
+	}
 	return manager.ListOpenWorkflowExecutions(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) ListClosedWorkflowExecutions(
 	ctx context.Context,
 	request *ListWorkflowExecutionsRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:    "list",
+		isOpen:       false,
+		domainName:   request.Domain,
+		workflowType: "",
+		workflowID:   "",
+		closeStatus:  math.MinInt,
+		customQuery:  "",
+	}, request.Domain)
 	manager := v.chooseVisibilityManagerForRead(request.Domain)
+	if override := ctx.Value("visibility-override"); override == Primary {
+		manager = v.esVisibilityManager
+	} else if override == Secondary {
+		manager = v.pinotVisibilityManager
+	}
 	return manager.ListClosedWorkflowExecutions(ctx, request)
 }
 
 func (v *pinotVisibilityTripleManager) ListOpenWorkflowExecutionsByType(
 	ctx context.Context,
 	request *ListWorkflowExecutionsByTypeRequest,
 ) (*ListWorkflowExecutionsResponse, error) {
+	v.logUserQueryParameters(userParameters{
+		operation:    "list",
```

Comment:
It is better to put the operations as enum and cast to string when you log it. At least you could use constants here.